sirepo/template/silas.py

```diff
@@ -38,64 +31,64 @@ def background_percent_complete(report, run_dir, is_running):
         percentComplete=0,
         frameCount=0,
     )
-    assert report == "crystalAnimation"
-    count = 0
-    path = run_dir.join(_CRYSTAL_CSV_FILE)
-    if path.exists():
-        with pkio.open_text(str(path)) as f:
-            for line in f:
-                count += 1
-        # first two lines are axis points
-        if count > 2:
-            plot_count = int((count - 2) / 2)
-            res.frameCount = plot_count
-            res.percentComplete = (
-                plot_count
-                * 100
-                / (
-                    1
-                    + data.models.crystalSettings.steps
-                    / data.models.crystalSettings.plotInterval
-                )
-            )
-    return res
-
-
-def post_execution_processing(success_exit, run_dir, **kwargs):
-    if success_exit:
-        return None
-    return _parse_silas_log(run_dir)
+    if report == "beamlineAnimation":
+        return _beamline_animation_percent_complete(run_dir, res, data)
+    if report in _SIM_DATA.SOURCE_REPORTS:
+        return _initial_intensity_percent_complete(
+            run_dir, res, data, _SIM_DATA.SOURCE_REPORTS
+        )
+    return _crystal_animation_percent_complete(run_dir, res, data)
 
 
 def get_data_file(run_dir, model, frame, options):
     if model in ("plotAnimation", "plot2Animation"):
         return _CRYSTAL_CSV_FILE
     if model == "crystal3dAnimation":
         return "intensity.npy"
-    if model in (
-        "initialIntensityReport",
-        "initialPhaseReport",
-    ) or _SIM_DATA.is_watchpoint(model):
-        return _RESULTS_FILE
+    if model == "beamlineAnimation0" or model in _SIM_DATA.SOURCE_REPORTS:
+        return _RESULTS_FILE.format(0)
+    if "beamlineAnimation" in model:
+        data = simulation_db.read_json(run_dir.join(template_common.INPUT_BASE_NAME))
+        for r in _beamline_animation_percent_complete(
+            run_dir, PKDict(frameCount=0), data
+        ).outputInfo:
+            if r.modelKey == model:
+                return r.filename
     raise AssertionError("unknown model={}".format(model))
 
 
+def post_execution_processing(success_exit, run_dir, **kwargs):
+    if success_exit:
+        return None
+    return _parse_silas_log(run_dir)
+
+
 def python_source_for_model(data, model, qcall, **kwargs):
     if model in ("crystal3dAnimation", "plotAnimation", "plot2Animation"):
         data.report = "crystalAnimation"
     else:
-        data.report = model or _last_watchpoint(data) or "initialIntensityReport"
+        if model:
+            model = re.sub("beamlineAnimation", "watchpointReport", model)
```

Comment:
I think the beamline logic was originally dealing with watchpointReport, but then SRW was changed to run as an animation, and silas also runs as an animation. You can try removing this, but I think when you select python-source on the plots, it may require this.